core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.spec.ts

```diff
@@ -554,6 +603,61 @@ describe('TopNavigationBarComponent', () => {
           AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH));
     }));
 
+  it('should fetch the number of unread feedback' +
+  'when user is logged In', fakeAsync(() => {
+    let userInfo = new UserInfo(
+      ['USER_ROLE'], true, false, false, false, true,
+      'en', 'username1', 'tester@example.com', true
+    );
+
+    spyOn(component, 'truncateNavbar').and.stub();
+    spyOn(userService, 'getUserInfoAsync').and.resolveTo(userInfo);
+    const fetchDataSpy = spyOn(
+      feedbackUpdatesBackendApiService,
+      'fetchFeedbackUpdatesDataAsync').and.returnValue(Promise.resolve({
+      numberOfUnreadThreads: FeedbackUpdatesData.
+        number_of_unread_threads,
+      threadSummaries: (
+        FeedbackUpdatesData.thread_summaries.map(
+          threadSummary => FeedbackThreadSummary
+            .createFromBackendDict(threadSummary))),
+      paginatedThreadsList: []
+    }));
+    component.userIsLoggedIn = true;
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.unreadThreadsCount).toBe(10);
+    expect(fetchDataSpy).toHaveBeenCalled();
+  }));
+
+  it('should show an alert when fails to' +
+  'get the feedback updates data', fakeAsync(() => {
+    let userInfo = new UserInfo(
+      ['USER_ROLE'], true, false, false, false, true,
+      'en', 'username1', 'tester@example.com', true
+    );
+
+    spyOn(component, 'truncateNavbar').and.stub();
+    spyOn(userService, 'getUserInfoAsync').and.resolveTo(userInfo);
+    const fetchDataSpy = spyOn(
+      feedbackUpdatesBackendApiService,
+      'fetchFeedbackUpdatesDataAsync')
+      .and.rejectWith(404);
+    const alertsSpy = spyOn(alertsService, 'addWarning').and.callThrough();
+
+    component.userIsLoggedIn = true;
+    component.ngOnInit();
+    tick(1010);
```

Comment:
Where did you get this number from?

Reply:
Earlier I have not used flush at the end of this test and it gives an 2 timer error, so I have used two tick one here and one at the end of test. But now I have used flush at the end of this test and remove the number from the tick().